### PR TITLE
Fix for metadata uploader overwriting existing data

### LINKED ISSUE #1488: Uploading metadata from a file removes pipeline-derived metadata from a project
## Describe the bug
If samples in a project have pipeline-derived metadata written to them and you upload a new metadata file, the pipeline-derived metadata gets deleted from the sample. See the below image for an example:

image

## Steps to reproduce the problem
What were you doing when you encountered the problem?
1. Start up an instance of IRIDA and run a pipeline to save metadata to the linelist/metadata table (e.g., AMR Detection).
2. Upload a metadata file with data for samples that had pipeline-derived results.
3. The pipeline-derived metadata entries will be removed and only the uploaded metadata fields will exist for the matching samples.

## Expected behaviour
I expected the pipeline-derived metadata to exist and be merged together with uploaded metadata.

## Additional context
This issue was tested on IRIDA 21.03.2.

The CSV file of metadata for the example screenshot: [meta-test.csv](https://github.com/phac-nml/irida/files/11335285/meta-test.csv)

### PR DESCRIPTION
**Pull request by ksierks**
## Description of changes
Fixing a bug where using the metadata uploader deleted exiting sample metadata. 

## Related issue
Fixes #1488.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* ~[ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Review by joshsadam — CHANGES_REQUESTED** [review]
This definitely works as advertised.  My only request is that you add a test to confirm the number of metadata fields on a sample pre and post file upload.

**Comment by ksierks** [comment]
> This definitely works as advertised.  My only request is that you add a test to confirm the number of metadata fields on a sample pre and post file upload.

Please see latest commit for better test.

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectSampleMetadataImportPageIT.java [hunk 1]
```diff
@@ -189,4 +192,25 @@ public void testFailedUploadByRemovingPrivileges() {
...
+		samplePage.clickSampleName(SAMPLE_NAME);
+		SampleDetailsViewer sampleDetailsViewer = SampleDetailsViewer.getSampleDetails(driver());
+		sampleDetailsViewer.clickMetadataTabLink();
+		assertTrue(sampleDetailsViewer.addNewMetadataButtonVisible());
+		sampleDetailsViewer.getNumberOfMetadataEntries();
+		assertEquals(10, sampleDetailsViewer.getNumberOfMetadataEntries(),
+				"Should have the proper number of metadata entries");
+		assertEquals("Sneezing", sampleDetailsViewer.getValueForMetadataField("symptom"),
+				"Should have existing metadata");
+		assertEquals("AB", sampleDetailsViewer.getValueForMetadataField("Province"), "Should have uploaded metadata");
```
Comment (joshsadam) [inline src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectSampleMetadataImportPageIT.java:214]: This looks great, thanks.

**Review by joshsadam — COMMENTED** [review]
This looks great.  Double-check the failing test.

**Review by joshsadam — APPROVED** [review]
Thanks for updating the test.

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**